packages/realm-web/src/utils/string.ts

```diff
@@ -35,15 +35,29 @@ export function generateRandomString(length: number, alphabet: string) {
  * Encode an object mapping from string to string, into a query string to be appended a URL.
  *
  * @param params The parameters to include in the string.
- * @returns A URL encoded representation of the parameters.
+ * @param prefixed Should the "?" prefix be added if values exists?
+ * @returns A URL encoded representation of the parameters (omitting a "?" prefix).
  */
-export function encodeQueryString(params: {
-    [key: string]: string | number | boolean;
-}) {
-    return Object.entries(params)
-        .map(([k, v]) => [k, encodeURIComponent(v)])
-        .map(([k, v]) => `${k}=${v}`)
-        .join("&");
+export function encodeQueryString(
+    params: {
+        [key: string]: string | number | boolean | undefined;
+    },
+    prefixed = true,
+) {
+    // Filter out undefined values
+    const entriesWithValue = Object.entries(params).filter(
+        ([k, v]) => typeof v !== "undefined",
+    ) as Array<[string, string | number | boolean]>;
+    // Determine if a prefixed "?" is appropreate
+    const prefix = prefixed && entriesWithValue.length > 0 ? "?" : "";
+    // Transform keys and values to a query string
+    return (
+        prefix +
+        entriesWithValue
+            .map(([k, v]) => [k, encodeURIComponent(v)])
+            .map(([k, v]) => `${k}=${v}`)
```

Comment:
Any reason to map twice? Seems like this could, just as easily & readable , be done in one.

Reply:
I agree.